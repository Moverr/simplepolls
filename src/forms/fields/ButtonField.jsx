import React from 'react'



function ButtonField(props){
    
    return <button {...props} onClick={props.callback}> props.title </button>;
}
export default ButtonField;