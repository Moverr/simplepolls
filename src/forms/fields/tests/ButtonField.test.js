import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
 
import ButtonField from '../ButtonField';

 
 const clickAlert  = jest.fn();
  
let wrapper = null;
beforeAll(()=>{

const props = {
        title:"Button Title",
        name:'Button Name',
        message:'Button Message',
        className:'buttonField',
        id:"buttonID",
        callback:clickAlert()
    };
    wrapper = shallow(<ButtonField props />); 
  
});

describe('Testing Button FIeld', () => {
 
  it('should  render correctly', () => {     
    expect(wrapper).toMatchSnapshot(); 
     });


  it('button should respond to click event ', () => {     
      expect(clickAlert).toHaveBeenCalled();      
      wrapper.find('button').simulate('click');       
      expect(clickAlert).toHaveBeenCalledTimes(1);

      
     });



})



 