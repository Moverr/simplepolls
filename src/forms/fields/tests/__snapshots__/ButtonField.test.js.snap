// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button FIeld should  render correctly 1`] = `
<button
  props={true}
>
   props.title 
</button>
`;
